opencti-platform/opencti-front/src/private/components/common/form/ObjectMarkingField.tsx

```diff
@@ -76,16 +115,67 @@ const ObjectMarkingField: FunctionComponent<ObjectMarkingFieldProps> = ({
   helpertext,
   disabled,
   label,
+  allowedMarkingOwnerId,
   setFieldValue,
   limitToMaxSharing = false,
   filterTargetIds,
+  isOptionEqualToValue,
 }) => {
   const classes = useStyles();
   const { t_i18n } = useFormatter();
   const [newMarking, setNewMarking] = useState<
   Option[] | OptionValues | undefined
   >(undefined);
   const [operation, setOperation] = useState<string | undefined>(undefined);
+  const [otherUserAllowedMarkingsData, setOtherUserAllowedMarkingsData] = useState(
+    [] as { definition: string | null | undefined, id: string, x_opencti_color: string | null | undefined }[] | undefined,
+  );
+
+  const fetchCreatorAllowedMarking = async (creatorId: string) => {
+    return fetchQuery(objectMarkingFieldAllowedMarkingQuery, {
+      id: creatorId,
+    })
+      .toPromise();
+  };
+
+  const fetchCreatorAllowedMarkings = async (creatorAllowedMarkingIds: string[]) => {
+    if (creatorAllowedMarkingIds.length) {
+      return fetchQuery(objectMarkingFieldOtherUserAllowedMarkingsQuery, {
+        filters: {
+          mode: 'and',
+          filters: {
+            mode: 'or',
+            key: 'id',
+            values: creatorAllowedMarkingIds,
+            operator: 'eq',
+          },
+          filterGroups: [],
+        },
+      })
+        .toPromise();
+    }
+    return {
+      markingDefinitions: {
+        edges: [],
+      },
+    };
+  };
+
+  useEffect(() => {
+    if (allowedMarkingOwnerId) {
+      const fetchData = async () => {
+        const creatorAllowedMarkingIds = ((await fetchCreatorAllowedMarking(allowedMarkingOwnerId) as unknown as ObjectMarkingFieldAllowedMarkingQuery$data)
+          .user?.groups?.edges ?? [])
+          .flatMap((group) => (group?.node?.allowed_marking ?? [])
+            .map((marking) => marking.id));
+        const markingsData = ((await fetchCreatorAllowedMarkings(creatorAllowedMarkingIds) as unknown as ObjectMarkingFieldOtherUserAllowedMarkingsQuery$data)
+          ?.markingDefinitions?.edges
+          .map((marking) => ({ ...marking.node })));
+        setOtherUserAllowedMarkingsData(markingsData);
+      };
+      fetchData().then();
```

Comment:
No need for then()

Reply:
The promise must be resolved.
Otherwise it is ignored.
![image](https://github.com/OpenCTI-Platform/opencti/assets/13460700/d2f3e581-5f02-4898-b6fc-1ee145413f2f)

Adding async/await break the useEffect signature.

Reply:
Is this an error of IntelliJ or check-ts or eslint ?

Reply:
IntelliJ.
Let me remove it then.